### DISCUSSION TITLE
Create submenu page

### DISCUSSION THREAD

**Discussion by vikasbhvsr** [jeffreyvr/wp-settings#20] [Q&A]
Hello,

I have created a admin menu page at the root. I have a requirement to create a submenu page with the parent of what I just created using WP Settings. 

Is there a way? am I missing something? Here's what my code looks like.

```php
 public function __construct()
    {
        add_action('admin_menu', [$this, 'ghm_settings_menu_page']);
    }

public function ghm_settings_menu_page()
    {
        $settings = new WPSettings(__('Settings'));

        $this->abc($settings);
        $this->xyz($settings);

        $settings->set_capability('manage_options');
        $settings->set_option_name('my_options');
        $settings->set_menu_icon('data:image/svg+xml;base64,customicon');
        $settings->make();
    }
```

**Comment by jeffreyvr** [comment]
At the moment you would have to add another instance of the `WPSettings` class, and then set your existing settings page as the parent through `$settings-> set_menu_parent_slug('your-parent-slug)`.

Have not tested it yet, but it should work.

**Reply by vikasbhvsr** [reply]
@jeffreyvr apologies on the delay. Work got in the way.

Unfortunately, your proposed solution does not work.
